### PR TITLE
My profile

### PR DESCRIPTION
**Pull request by weezyjr**
```

My Profile: Make sure after you change the password and press on cancel button, ... 
2019-12-31 17:31:37.370 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test001_user_can_change_password_and_press_on_cancel
2019-12-31 17:31:54.270 | INFO     | api_testing.apis.base_api:info:68 - Get authorized api session.
2019-12-31 17:31:56.149 | INFO     | api_testing.apis.base_api:info:68 - session ID : eyJhbGciOi .....
2019-12-31 17:31:56.150 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : profile_test.
2019-12-31 17:32:32.351 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:32:35.324 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:32:40.329 | INFO     | ui_testing.pages.my_profile_page:change_password:16 -  + Change the password
2019-12-31 17:32:51.241 | INFO     | ui_testing.pages.my_profile_page:change_password:31 -  + New password Canceled
2019-12-31 17:32:51.242 | INFO     | ui_testing.pages.header_page:click_on_header_button:25 - Press on the header button
2019-12-31 17:32:51.414 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:32:56.637 | INFO     | ui_testing.pages.header_page:logout:32 - + Logout
2019-12-31 17:32:56.637 | INFO     | ui_testing.pages.base_pages:sleep_medium:43 -  Medium sleep.
2019-12-31 17:33:06.644 | INFO     | api_testing.apis.base_api:info:68 - Get authorized api session.
2019-12-31 17:33:08.333 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 
ok
My Profile: Make sure that the user name & email displayed above the language ... 
2019-12-31 17:33:08.334 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test002_my_profile_should_show_username_and_email
2019-12-31 17:33:25.312 | INFO     | api_testing.apis.base_api:info:68 - Get authorized api session.
2019-12-31 17:33:25.974 | INFO     | api_testing.apis.base_api:info:68 - session ID : eyJhbGciOi .....
2019-12-31 17:33:25.976 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : profile_test.
2019-12-31 17:33:55.558 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:33:58.513 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:34:03.685 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 
ok
My Profile: Make sure that you can change the password ... 
2019-12-31 17:34:03.686 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test003_user_can_change_password_and_login_successfully
2019-12-31 17:34:20.485 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : profile_test.
2019-12-31 17:34:55.428 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:34:58.367 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:35:03.371 | INFO     | ui_testing.pages.my_profile_page:change_password:16 -  + Change the password
2019-12-31 17:35:03.886 | INFO     | ui_testing.pages.base_pages:save:51 - save the changes
2019-12-31 17:35:13.993 | INFO     | ui_testing.pages.my_profile_page:change_password:28 -  + New password saved
2019-12-31 17:35:13.993 | INFO     | ui_testing.pages.header_page:click_on_header_button:25 - Press on the header button
2019-12-31 17:35:14.099 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:35:19.272 | INFO     | ui_testing.pages.header_page:logout:32 - + Logout
2019-12-31 17:35:19.273 | INFO     | ui_testing.pages.base_pages:sleep_medium:43 -  Medium sleep.
2019-12-31 17:35:29.282 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : 20b8657dc4.
2019-12-31 17:35:33.401 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:35:35.521 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:35:40.526 | INFO     | ui_testing.pages.my_profile_page:change_password:16 -  + Change the password
2019-12-31 17:35:41.063 | INFO     | ui_testing.pages.base_pages:save:51 - save the changes
2019-12-31 17:35:51.208 | INFO     | ui_testing.pages.my_profile_page:change_password:28 -  + New password saved
2019-12-31 17:35:51.276 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 
ok
My Profile: Language Approach: Make sure that you can change language [with lang='EN'] ... 
2019-12-31 17:35:51.277 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test004_user_can_change_the_language_0_EN
2019-12-31 17:36:08.076 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : profile_test.
2019-12-31 17:36:30.991 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:36:33.740 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:36:38.742 | INFO     | ui_testing.pages.my_profile_page:chang_lang:34 -  + Change language to EN
2019-12-31 17:36:41.074 | INFO     | ui_testing.pages.base_pages:sleep_large:47 -  Large sleep.
2019-12-31 17:36:56.213 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 
ok
My Profile: Language Approach: Make sure that you can change language [with lang='DE'] ... 
2019-12-31 17:36:56.214 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test004_user_can_change_the_language_1_DE
2019-12-31 17:37:12.985 | INFO     | ui_testing.pages.login_page:login:10 - Login profile_test : profile_test.
2019-12-31 17:37:40.463 | INFO     | ui_testing.pages.my_profile_page:get_my_profile_page:11 -  + Get my profile page.
2019-12-31 17:37:43.477 | INFO     | ui_testing.pages.base_pages:sleep_small:39 -  Small sleep.
2019-12-31 17:37:48.480 | INFO     | ui_testing.pages.my_profile_page:chang_lang:34 -  + Change language to DE
2019-12-31 17:37:50.783 | INFO     | ui_testing.pages.base_pages:sleep_large:47 -  Large sleep.
2019-12-31 17:38:05.854 | INFO     | ui_testing.pages.my_profile_page:chang_lang:34 -  + Change language to EN
2019-12-31 17:38:08.249 | INFO     | ui_testing.pages.base_pages:sleep_large:47 -  Large sleep.
2019-12-31 17:38:23.330 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 
ok

----------------------------------------------------------------------
Ran 5 tests in 405.961s

OK

```

### CODE REVIEW

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        self.header_page.logout()
+
+        # try to authorize with the new password
+        try:
+            baseAPI = BaseAPI()
+            baseAPI._get_authorized_session(username=self.base_selenium.username, password=self.new_password, reset_token=True)
+        except:
+            failed_to_login = True
+        finally:
+            self.assertTrue(failed_to_login)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: in any case, this gonna pass

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    def test002_my_profile_should_show_username_and_email(self):
+        """
+        My Profile: Make sure that the user name & email displayed above the language
+
+        LIMS-6090
+        """
+        username = self.base_selenium.get_text(element='my_profile:username')
+        email = self.base_selenium.get_text(element='my_profile:email')
+        self.assertTrue(username)
+        self.assertTrue(email)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: assertEqual

File: .travis.yml [hunk 3]
```diff
@@ -21,10 +21,9 @@ install:
...
-  - export TESTS='test07_company_profile.py:'
+  - export TESTS='test09_my_profile.py:'
   - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then export TESTS=''; fi
   - echo $TESTS
   - cd $TRAVIS_BUILD_DIR && export PYTHONPATH='./'
-#  - export TESTS='test04_testunits.py:TestUnitsTestCases.test010_material_type_approach_1_Quantitative_MiBi'
-#  - xvfb-run -a nosetests-3.4 -vs --nologcapture ui_testing/testcases/basic_tests/$TESTS --tc-file=config.ini
-  - xvfb-run -a nosetests-3.4 -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test05_contacts.py:ContactsTestCases.test015_create_user_with_role_contact
+  # - xvfb-run -a nosetests-3.4 -vs --nologcapture ui_testing/testcases/header_tests/$TESTS --tc-file=config.ini
+  - xvfb-run -a nosetests-3.4 -vs --nologcapture --tc-file=config.ini ui_testing/testcases/header_tests/test09_my_profile.py --tc=site.username:my_profile --tc=site.password:my_profile
```
[the remark this answers is not in the document]
Comment (weezyjr) [reply .travis.yml]: @0xIslamTaha that branch containing the user API is not merged yet.

[the remark this answers is not in the document]
Comment (0xIslamTaha) [reply .travis.yml]: U have it now 
https://github.com/modeso-open-source/1lims-automation/blob/4c0f28424c96a35ad7292b86e6d7423a78911520/api_testing/apis/users_api.py#L5

[the remark this answers is not in the document]
Comment (weezyjr) [reply .travis.yml]: @0xIslamTaha Used it as soons as it was merged, you can review it now :+1:

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+        # init the flags
+        self.reset_language = False
+
+    def tearDown(self):
+        # reset the english language
+        if self.reset_language:
+            self.my_profile_page.chang_lang('EN')
+            self.reset_language = False
+        return super().tearDown()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: Update the tearDown regarding to the last comment.

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+    def test001_user_can_change_password_and_press_on_cancel(self):
+        """
+        My Profile: Make sure after you change the password and press on cancel button, 
+        the password shouldn't change
+
+        LIMS-6091
+        """
+        # new password value
+        self.new_password = self.my_profile_page.generate_random_text()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]:
```suggestion
        new_password = self.my_profile_page.generate_random_text()
```

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        self.my_profile_page.change_password(self.current_password, self.new_password)
+
+        # try to authorize with the new password
+        try:
+            baseAPI = BaseAPI()
+            baseAPI._get_authorized_session(username=self.base_selenium.username, password=self.new_password, reset_token=True)
+        except:
+            failed_to_login = True
+        else:
+            raise Exception('The user was able to authorize with an unsaved password')
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: it should be failed not raising an exception

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        """
+        My Profile: Make sure after you change the password and press on cancel button, 
+        the password shouldn't change
+
+        LIMS-6091
+        """
+        # new password value
+        self.new_password = self.my_profile_page.generate_random_text()
+        # flag to insure that the user failed to login
+        failed_to_login = False
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: remove this line

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        # try to authorize with the new password
+        try:
+            baseAPI = BaseAPI()
+            baseAPI._get_authorized_session(username=self.base_selenium.username, password=self.new_password, reset_token=True)
+        except:
+            failed_to_login = True
+        else:
+            raise Exception('The user was able to authorize with an unsaved password')
+        finally:
+            self.assertTrue(failed_to_login)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]:
```
baseAPI = BaseAPI()
self.assertRaises(EXCEPTION, baseAPI._get_authorized_session(username=self.base_selenium.username, password=self.new_password, reset_token=True)
```

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+    def test003_user_can_change_password_and_login_successfully(self):
+        """
+        My Profile: Make sure that you can change the password 
+        and login with the new one successfully 
+
+        LIMS-6084
+        """
+        # new password value
+        self.new_password = self.my_profile_page.generate_random_text()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]:
```suggestion
        new_password = self.my_profile_page.generate_random_text()
```

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+        LIMS-6089
+        """
+        # change language
+        self.my_profile_page.chang_lang(lang)
+
+        # change the EN to DE
+        self.my_profile_page.chang_lang('DE')
+        # to restore the language
+        self.reset_language = True
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: remove it

File: ui_testing/testcases/header_tests/test09_my_profile.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+        
+        if lang == 'EN':
+            # change the DE to EN
+            self.my_profile_page.chang_lang('EN')
+            # reset the flag
+            self.reset_language = False
```
Comment (0xIslamTaha) [inline ui_testing/testcases/header_tests/test09_my_profile.py]: remove it

File: api_testing/apis/base_api.py [hunk 12]
```diff
@@ -31,18 +31,28 @@ def __init__(self):
                         'cache-control': "no-cache"}
         self._get_authorized_session()
 
-    def _get_authorized_session(self):
+    def _get_authorized_session(self, username=None, password=None, reset_token=False):
+        if not username:
+            username = self.username
```
[the remark this answers is not in the document]
Comment (weezyjr) [reply api_testing/apis/base_api.py]: What is the point of this change ?

**Pull request merged**